conda_build/cli/main_debug.py

```diff
@@ -43,51 +42,46 @@ def parse_args(args):
                          "The top-level recipe can be specified by passing 'TOPLEVEL' here"))
     p.add_argument("-a", "--activate-string-only", action="store_true",
                    help="Output only the string to the used generated activation script.  Use this for creating envs in scripted "
-                   "environments.")
+                        "environments.")
 
     # cut out some args from render that don't make sense here
     #    https://stackoverflow.com/a/32809642/1170370
     p._handle_conflict_resolve(None, [('--output', [_ for _ in p._actions if _.option_strings == ['--output']][0])])
     p._handle_conflict_resolve(None, [('--bootstrap', [_ for _ in p._actions if _.option_strings == ['--bootstrap']][0])])
     p._handle_conflict_resolve(None, [('--old-build-string', [_ for _ in p._actions if
                                                               _.option_strings == ['--old-build-string']][0])])
-    args = p.parse_args(args)
-    return p, args
+    return p
 
 
-def execute(args):
-    p, _args = parse_args(args)
-    test = True
+def execute():
+    parser = get_parser()
+    args = parser.parse_args()
 
     try:
-        if not any(os.path.splitext(_args.recipe_or_package_file_path)[1] in ext for ext in CONDA_PACKAGE_EXTENSIONS):
-            # --output silences console output here
-            thing_to_debug = render_execute(args, print_results=False)
-            test = False
-        else:
-            thing_to_debug = _args.recipe_or_package_file_path
-        activation_string = api.debug(thing_to_debug, verbose=(not _args.activate_string_only), **_args.__dict__)
+        activation_string = api.debug(
+            args.recipe_or_package_file_path,
+            verbose=(not args.activate_string_only),
+            **args.__dict__
+        )
 
-        if not _args.activate_string_only:
+        if not args.activate_string_only:
             print("#" * 80)
-            if test:
-                print("Test environment created for debugging.  To enter a debugging environment:\n")
-            else:
-                print("Build and/or host environments created for debugging.  To enter a debugging environment:\n")
+            print(
+                "Build and/or host environments created for debugging.  To enter a debugging environment:\n"
+            )
```

Comment:
I think we grabbed the wrong if-clause? We've kept the else-clause from above:

```python
else:
    thing_to_debug = _args.recipe_or_package_file_path
```

Meaning `test = True` so we want to keep the first if-clause here:

```python
print("Test environment created for debugging. To enter a debugging environment:\n")
```